ts.js

```diff
@@ -20,6 +20,9 @@ module.exports = {
     '@typescript-eslint/explicit-function-return-type': 'off', // allow compiler to derive return type
     '@typescript-eslint/no-this-alias': 'off', // allow 'const self = this'
     'jsdoc/require-param': 'off', // do not require jsdoc for params
-    'jsdoc/require-param-type': 'off' // allow compiler to derive param type
+    'jsdoc/require-param-type': 'off', // allow compiler to derive param type
+    '@typescript-eslint/no-misused-promises': 'error',
+    '@typescript-eslint/no-floating-promises': 'error',
```

Comment:
> '@typescript-eslint/no-floating-promises': 'error',

This will probably be disruptive as it's a common pattern to just `await` a thing if it may or may not be a promise.  Unnecessary async kills performance though so it's probably a good idea.

Given it will have to go out as a major, how about adding the rule to require declared return types too?

Reply:
Seems like other rules are included. Maybe we were encountering those issues because we used some older aegir 🤷🏻 
Should I change this PR to only include https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/await-thenable.md to help increase libp2p performance?

Reply:
Yes please!